### PR TITLE
Test setup

### PR DESCRIPTION
**Pull request by ViaMarcus**
## Setting up test environment
[PT](https://www.pivotaltracker.com/story/show/172761556)
Adding and initializing rspec
Adding and configuring shoulda-matchers
Adding and configuring factory-bot
Adding rails-pry
Adding coveralls
Cleaning files

### CODE REVIEW

**Comment by ViaMarcus** [comment]
Not sure why, but semaphore cannot track this PR. Perhaps because it was opened before semaphore was setup

File: spec/rails_helper.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
+require 'spec_helper'
+ENV['RAILS_ENV'] ||= 'test'
+require File.expand_path('../config/environment', __dir__)
+
+abort("The Rails environment is running in production mode!") if Rails.env.production?
+require 'rspec/rails'
+require 'coveralls'
```
Comment (oliverochman) [inline spec/rails_helper.rb]: line 7 and 8 need to be at the top of this file

File: spec/rails_helper.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+end
+
+Dir[Rails.root.join('spec/support/**/*.rb')].each { |f| require f }
+
+RSpec.configure do |config|
+  config.fixture_path = "#{::Rails.root}/spec/fixtures"
+  config.use_transactional_fixtures = true
+  config.infer_spec_type_from_file_location!
+  config.filter_rails_from_backtrace!
+  # config.filter_gems_from_backtrace("gem name")
```
Comment (oliverochman) [inline spec/rails_helper.rb:25]: remove commented out code

**Review by oliverochman — CHANGES_REQUESTED** [review]

File: spec/rails_helper.rb (hunk 2 above)
Comment (ViaMarcus) [reply spec/rails_helper.rb:25]: I wanted to keep this as it is very likely we will need to use this functionality once we start running tests. Keeping the comment here will make sure there are no struggles with the syntax once it is needed.

File: spec/rails_helper.rb (hunk 1 above)
Comment (ViaMarcus) [reply spec/rails_helper.rb]: Moved them to top, fixed the coveralls wear to a coveralls wear merged

**Pull request merged**